python/pyspark/sql/types.py

```diff
@@ -186,6 +187,20 @@ def fromInternal(self, ts):
             return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
 
 
+class TimestampNTZType(TimestampType):
```

Comment:
@dominikgehl can we do this in a separate PR? With adding this type, we should also check if it works w/:
- `SparkSession.createDataFrame` (when 'spark.sql.execution.arrow.pyspark.enabled'` configuration is on and off)
- `DataFrame.toPandas` (when 'spark.sql.execution.arrow.pyspark.enabled'` configuration is on and off)
- `DataFrame.collect()`
- `pyspark.functions.udf`
- `pyspark.functions.pandas_udf`

Reply:
@HyukjinKwon Just to get the workflow right, should I create a separate JIRA ticket to create the TimestampNTZType ? And then once we have the additional type, we come back to this PR ?

Reply:
Yes, let;s create a new ticket for `TimestampNTZType`. Yes, let's do that.

Reply:
New ticket is https://issues.apache.org/jira/projects/SPARK/issues/SPARK-36287